linux/network_device_ops/enable_new_ethernet.yml

```diff
@@ -57,53 +57,88 @@
     - debug: var=network_config.stdout_lines
   when: guest_os_ansible_distribution not in ["VMware Photon OS", "Astra Linux (Orel)"]
 
-# Apply netplan network config changes in Ubuntu server
+# Check new ethernet device status to see whether it has been bring up automatically
+- include_tasks: ../utils/ethernet_ip_get.yml
+  vars:
+    all_ip: False
+
+- name: "Set fact of ethernet device status and IP"
+  set_fact:
+    eth_dev_status: "{{ ethernets[eth_dev] if ethernets | length > 0 else {} }}"
+
 - block:
-    - name: "Apply netplan configuration file for new added nic {{ eth_dev }}"
-      command: "netplan apply"
+    - include_tasks: ../utils/get_service_info.yml
+      vars:
+        service_name: "NetworkManager.service"
+
+    # RHEL/SLED/Ubuntu Desktop network interfaces are managed by NetworkManager
+    - name: "Bring link '{{ eth_dev }}' up"
+      shell: "nmcli device connect {{ eth_dev }}"
       delegate_to: "{{ vm_guest_ip }}"
-      become: yes
-    - include_tasks: ../../common/vm_wait_connection.yml
-  when:
-    - guest_os_ansible_distribution == "Ubuntu"
-    - guest_os_with_gui is defined and not guest_os_with_gui
-
-# Bring up the new network interface
-- name: "Bring link '{{ eth_dev }}' up"
-  shell: "nmcli device connect {{ eth_dev }}"
-  delegate_to: "{{ vm_guest_ip }}"
-  register: set_link_down_up
-  when:
-    - guest_os_family == "RedHat"
-    - guest_os_ansible_distribution_major_ver | int >= 8
-
-- name: "Bring link '{{ eth_dev }}' up"
-  shell: "ip link set {{ eth_dev }} down; ip link set {{ eth_dev }} up"
-  delegate_to: "{{ vm_guest_ip }}"
-  register: set_link_down_up
-  when: >
-    (guest_os_ansible_distribution in ["VMware Photon OS", "Flatcar"]) or
-    (guest_os_ansible_distribution == "Ubuntu" and guest_os_with_gui is defined and guest_os_with_gui)
```

Comment:
Is this situation also be covered? `(guest_os_ansible_distribution == "Ubuntu" and guest_os_with_gui is defined and guest_os_with_gui)` by Network manager or "netplan"

Reply:
Yes. Ubuntu Desktop is using Network Manager, Ubuntu server is using netplan.

```
Ubuntu  21.10 Desktop
+---------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                |
|                           | bitness='64'                      |
|                           | distroName='Ubuntu'               |
|                           | distroVersion='21.10'             |
|                           | familyName='Linux'                |
|                           | kernelVersion='5.13.0-22-generic' |
|                           | prettyName='Ubuntu 21.10'         |
+---------------------------------------------------------------+


Test Results (Total: 6, Failed: 1, No Run: 0, Elapsed Time: 01:29:48):
+---------------------------------------------------+
| Name                       |   Status | Exec Time |
+---------------------------------------------------+
| e1000e_network_device_ops  |   Passed | 00:06:03  |
| vmxnet3_network_device_ops |   Passed | 00:03:01  |
```
```
Ubuntu 21.10 Live Server
+---------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                |
|                           | bitness='64'                      |
|                           | distroName='Ubuntu'               |
|                           | distroVersion='21.10'             |
|                           | familyName='Linux'                |
|                           | kernelVersion='5.13.0-22-generic' |
|                           | prettyName='Ubuntu 21.10'         |
+---------------------------------------------------------------+


Test Results (Total: 2, Failed: 0, No Run: 0, Elapsed Time: 00:05:46):
+-------------------------------------------------+
| Name                       | Status | Exec Time |
+-------------------------------------------------+
| e1000e_network_device_ops  | Passed | 00:02:05  |
| vmxnet3_network_device_ops | Passed | 00:01:45  |
+-------------------------------------------------+
```